core/frontend-stubs/cli-frontend-stub/src/main/java/org/eclipse/sw360/antenna/frontend/stub/cli/AbstractAntennaCLIOptions.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) Bosch.IO GmbH 2020.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.frontend.stub.cli;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public abstract class AbstractAntennaCLIOptions {
+    /**
+     * The prefix to identify command line switches.
+     */
+    public static final String SWITCH_PREFIX = "-";
+
+    /**
+     * The short command line switch to enable debug logging.
+     */
+    public static final String SWITCH_DEBUG_SHORT = SWITCH_PREFIX + "X";
+
+    /**
+     * The long command line switch to enable debug logging.
+     */
+    public static final String SWITCH_DEBUG_LONG = SWITCH_PREFIX + "-debug";
+
+    /**
+     * The short command line switch to display a help message.
+     */
+    public static final String SWITCH_HELP_SHORT = SWITCH_PREFIX + "h";
+
+    /**
+     * The long command line switch to display a help message.
+     */
+    public static final String SWITCH_HELP_LONG = SWITCH_PREFIX + "-help";
+
+    /**
+     * Flag whether debug logging should be enabled.
+     */
+    protected final boolean debugLog;
+
+    /**
+     * Flag whether the help message should be printed.
+     */
+    protected final boolean showHelp;
+
+    /**
+     * Flag whether the command line could be parsed successfully.
+     */
+    protected final boolean valid;
+
+    public AbstractAntennaCLIOptions(boolean debugLog, boolean showHelp, boolean valid) {
```

Comment:
Minor minor nit: For abstract classes it makes sense to make the constructor protected.